mlflow/tracking/client.py

```diff
@@ -433,6 +440,223 @@ def get_trace_info(self, trace_id: str) -> TraceInfo:
         """
         return self._tracking_client.get_trace_info(trace_id)
 
+    def start_trace(
+        self,
+        name: str,
+        inputs: Optional[Dict[str, Any]] = None,
+        attributes: Optional[Dict[str, str]] = None,
+        tags: Optional[Dict[str, str]] = None,
+    ):
+        """
+        Create a new trace object and start a root span under it.
+
+        This is an imperative API to manually create a new span under a specific trace id and parent
+        span, unlike the higher-level APIs like `@mlflow.trace()` and `with mlflow.start_span()`,
+        which automatically manage the span lifecycle and parent-child relationship.
+
+        A trace started with this method must be ended by calling `end_trace(trace_id)`.
+
+        Args:
+            name: The name of the trace (and the root span).
+            inputs: Inputs to set on the root span of the trace.
+            attributes: A dictionary of attributes to set on the root span of the trace.
+            tags: A dictionary of tags to set on the trace.
+
+        Returns:
+            An MLflowSpanWrapper object representing the root span of the trace.
+
+        Example:
+
+        .. code-block:: python
+
+                from mlflow.tracking import MlflowClient
+
+                client = MlflowClient()
+
+                root_span = client.start_trace("my_trace")
+                trace_id = root_span.trace_id
+
+                # Create a child span
+                child_span = client.start_span(
+                    "child_span", trace_id=trace_id, parent_span_id=root_span.span_id
+                )
+                # Do something...
+                client.end_span(trace_id=trace_id, span_id=child_span.span_id)
+
+                client.end_trace(trace_id)
+        """
+        trace_manager = InMemoryTraceManager.get_instance()
+        root_span = trace_manager.start_detached_span(name)
+
+        if inputs:
+            root_span.set_inputs(inputs)
+        if attributes:
+            root_span.set_attributes(attributes)
+
+        trace_info = trace_manager.get_trace_info(root_span.trace_id)
+        trace_info.tags.update(tags or {})
+
+        return root_span
+
+    def end_trace(
+        self,
+        trace_id: str,
+        inputs: Optional[Dict[str, Any]] = None,
```

Comment:
What is the resolution for duplicated inputs? 
If I set inputs on start_trace, and then call end_trace with the same inputs, do we resolve these by updating the dict? 
Broader question: should we allow `inputs` on `end_trace`? I feel like it might stifle some creative uses of this API, but removing it might make it less confusing for users who are using this imperative API.

Reply:
> If I set inputs on start_trace, and then call end_trace with the same inputs, do we resolve these by updating the dict?

The input will be overridden to the one specified in end_trace.

> Broader question: should we allow inputs on end_trace?

Great point. I just followed what Langfuse does (have input/output in both end_xxx), but personally I agree with you it's cleaner separation of responsibility if we remove `input` from end_xxx APIs. Let me do the change and see feedbacks from the preview.

Reply:
sounds good!